controllers/feeds/reactionList.js

```diff
@@ -28,6 +42,8 @@ module.exports = async (req, res) => {
       comment?.latest_children?.comment?.forEach(collectUserIds);
     };
     level1Comments.forEach(collectUserIds);
+    // Filter out comments from blocked users
+    level1Comments = level1Comments.filter((comment) => !blockedUsers.includes(comment.user.id));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This line of code filters out comments from blocked users, which is good. However, it does so by checking each comment's user id against the list of blocked users. This operation has a time complexity of O(n*m), where n is the number of comments and m is the number of blocked users. If both lists are large, this could lead to performance issues. 

A more efficient way would be to convert the `blockedUsers` array into a Set, which allows for constant-time lookup operations. Here's how you can do that:

```diff
    const blockedUsers = new Set(blockedUserArray.map((user) => user.user_id_blocked));
...
-   level1Comments = level1Comments.filter((comment) => !blockedUsers.includes(comment.user.id));
+   level1Comments = level1Comments.filter((comment) => !blockedUsers.has(comment.user.id));
```

This change reduces the time complexity of the filtering operation to O(n), which is a significant improvement for large lists.


<!-- This is an auto-generated comment by OSS CodeRabbit -->